lib/collector_syncronization.rb

```diff
@@ -0,0 +1,113 @@
+require 'uri'
+require 'securerandom'
+
+require 'global_configuration'
+require 'logging'
+require 'vsphere_session'
+require 'metrics_collector'
+require 'infrastructure_collector'
+require 'inventory_collector'
+require 'uc6_connector'
+require 'interval_time'
+
+class CollectorSyncronization
+  using IntervalTime
+  include GlobalConfiguration
+  include Logging
+  include VSphere
+
+  def initialize
+    @environment = ENV['METER_ENV'] || 'development'
+    @configuration = GlobalConfiguration::GlobalConfig.instance
+  end
+
+  def sync_data
+    start_sync if access_granted?
+  end
+
+  private
+
+  def access_granted?
+    @configuration[:verified_api_connection] && @configuration[:verified_vsphere_connection]
+  end
+
+  def start_sync
+    begin
+      logger.debug "On Start Sync"
```

Comment:
I would use single quotes here since there is no variable interpolation within the string